File: java/sort/BubbleSort.java
package sort;

import java.util.Arrays;

public class BubbleSort {
    public static void main(String[] args) {
        int[] array = {5, 2, 9, 1, 5, 6};
        bubbleSort(array);
        System.out.println(Arrays.toString(array));
    }

    public static void bubbleSort(int[] arr) {
        if (arr == null || arr.length <= 1) return;
        
        int n = arr.length;
        boolean swapped;
        
        for (int i = 0; i < n - 1; i++) {
            swapped = false;
            
            for (int j = 0; j < n - i - 1; j++) {
                if (arr[j] > arr[j + 1]) {
                    // Intercambiar arr[j] y arr[j+1]
                    int temp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = temp;
                    swapped = true;
                }
            }
            
            // Si no hubo intercambios en esta pasada, el array está ordenado
            if (!swapped) break;
        }
    }
}

File: python/sort/bubble_sort.py
def bubble_sort(arr):
    n = len(arr)
    
    for i in range(n):
        # Bandera para optimizar si el array ya está ordenado
        swapped = False
        
        for j in range(0, n - i - 1):
            if arr[j] > arr[j + 1]:
                arr[j], arr[j + 1] = arr[j + 1], arr[j]
                swapped = True
        
        # Si no hubo intercambios en esta pasada, el array está ordenado
        if not swapped:
            break
    
    return arr

if __name__ == "__main__":
    array = [5, 2, 9, 1, 5, 6]
    sorted_array = bubble_sort(array.copy())
    print(sorted_array)
